ivadomed/main.py

```diff
@@ -172,6 +174,24 @@ def run_main(config=None):
         with open(os.path.join(log_directory, "config_file.json"), 'w') as fp:
             json.dump(context, fp, indent=4)
 
+        # Model directory
+        path_model = os.path.join(log_directory, context["model_name"])
+        if not os.path.isdir(path_model):
+            print('Creating model directory: {}'.format(path_model))
+            os.makedirs(path_model)
+        else:
+            print('Model directory already exists: {}'.format(path_model))
+        # Copy model metadata
+        shutil.copyfile(os.path.join(log_directory, "config_file.json"),
+                        os.path.join(path_model, context["model_name"]+".json"))
+        # Copy model and convert it to ONNX format
+        shutil.copyfile(os.path.join(log_directory, "best_model.pt"),
+                        os.path.join(path_model, context["model_name"] + ".pt"))
+        convert_pytorch_to_onnx(fname_model=os.path.join(path_model, context["model_name"] + ".pt"),
```

Comment:
Here instead of saving, converting .pt file to .onnx, then deleting .pt file why not save it directly: `imed_utils.save_onnx_model(model, var_input, model_path.replace('pt', 'onnx'))`. This code could also be moved in function `train` to have access to the var_input variable.

Reply:
Yes, that's a bit of an overkill, I agree. The reason was: to do all this saving/packaging at the same place.
But I agree, I am going to move this to `train`.

Reply:
Ah and the other reason was: to avoid sending a new parameter to the training (`model_name`).. oh well

Reply:
Ah and last reason was: to avoid saving the best model twice (pt and onnx) at each epoch where we have a new best model.

Reply:
ahah at the end I am not too sure what is best

Reply:
The model doesn't have to be saved at every epoch, the saving + packaging can be put at the same place, at the end of `train` function after the `for` loop (so this covers, 2 out of 3 issues). As for the extra parameter, I agree it is not ideal. The model_name could be placed into `model_params` or `training_parameters` to avoid passing another param, but I don't know if it's better.

Another approach would be to leave the code where it is but create a dummy input (instead of using var_input).

I don't have a strong preference :) I let the choice up to you!